lib/core/notifier.js

```diff
@@ -277,30 +276,24 @@ class NotifierController {
    * @param {Request} request
    * @return {Promise}
    */
-  notifyDocumentUpdate (request) {
-    let
-      cacheId,
-      matchedRooms,
-      updatedDocument;
-
-    const { index, collection, _id } = request.input.resource;
-
-    // @todo do not make another get request, use the _source returned by update request
-    return this.storageEngine.get(index, collection, _id)
-      .then(result => {
-        updatedDocument = result;
-        matchedRooms = this._test(request, result._source, result._id);
-        cacheId = getCachePrefix(request) + updatedDocument._id;
-
-        const updatedFields = Object.keys(request.input.body)
-          .filter(_updatedFields => _updatedFields !== '_kuzzle_info');
-
-        return this.notifyDocument(matchedRooms, request, 'in', 'update', {
-          _id: updatedDocument._id,
-          _source: updatedDocument._source,
-          _updatedFields: updatedFields
-        });
-      })
+  notifyDocumentUpdate (request, updatedDocument) {
+
+    const matchedRooms = this._test(
+      request,
+      updatedDocument._source,
+      updatedDocument._id
+    );
+    const cacheId = getCachePrefix(request) + updatedDocument._id;
+
+    const updatedFields = Object.keys(request.input.body).filter(
```

Comment:
Nitpicking: according to our coding standards, the `.filter(...)`  should be on the new line